### ISSUE TITLE
[help] LFS repo size keep incresing as the exact size of the commite file instead of its delta

### ISSUE THREAD

**Issue by avatar-lavventura** [git-lfs/git-lfs#5380]
I am using `git-lfs` for pdf files (`git lfs track "*.pdf"`) that are keep edited and changed.

Afterwards,I observe that `Git-Repository` size remain same and after each commit LFS size increases as the exact size of the pdf file instead of only pdf's changed delta.

I was wondering is the complete pdf file is commited or its delta change commited into LFS?

**Comment by znull** [comment]
> #> I was wondering is the complete pdf file is commited or its delta change commited into LFS?

Yes, the complete file is committed each time. There is no delta (or other) compression applied to files in LFS. (#260)

 #> Is it wise to commit files into LFS that are edited over time?

That depends on the size of the files, how frequently they change, and how much space you're willing to use for them.

**Comment by bk2204** [comment]
I should also add that it's often the case that files stored in LFS (especially images, but often also PDFs) are already partially or mostly compressed.  As a result, trying to deltify or compress already-compressed files may actually expand them, which is why we don't do that by default, unlike Git.

I think this answers the question, so I'm going to close this, but let us know if you have more questions, and we can reopen.

**Comment by avatar-lavventura** [comment]
@bk2204 Thanks all for the input.

? how frequently they change?
I am working on my thesis file on a latex file. I also keep its pdf output few times a day. Overall pdf is around few MB but my delta-changes are usually few lines of code. 

You mentioned `we don't do that by default`, in future would it be considered to have an option option to enable to commit only for the `delta` changes, like for the cases of mine?

**Comment by avatar-lavventura** [comment]
@bk2204 I am the only one using a private git repository. I don't mind commiting build products to my repository.

**Comment by alexhaffner** [comment]
@bk2204 

I've been wondering for a a long time what would be a good way to manage game project files with git. There is often a bunch of binary assets part of the project, that still need to be versioned somehow. Would it be possible to create a "smart" or configurable system that does deltification, or compression, or nothing, depending on the file format and/or file characteristics? I think that would be a game changer for game dev projects.

**Comment by bk2204** [comment]
To add things like deltification or compression to objects, we're effectively requiring a smart server on the remote side, which isn't required right now.  For example, it's possible to use a server that does the LFS object protocol with a cloud bucket for the actual objects (which is in fact what GitHub does).  If we implement deltification and compression, then we're in fact implementing a lot of what Git already does, and it might be better to simply extend Git to support larger files better.

**Comment by stadlerb** [comment]
@avatar-lavventura Unrelated to the requested feature, but for your particular problem, you can turn off PDF compression (e.g. using `\pdfcompresslevel0` if you use pdflatex) and commit directly to Git instead of LFS.
